### PR TITLE
Javascritpt Capston Project

### PR DESCRIPTION
**Pull request by jaamanu**
Add functionality to likes button
Add test files
Create class object to allow comments 
Fix errors
fix linters
Add footer to app

### CODE REVIEW

File: src/modules/comments.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/* eslint-disable */
```
Comment (Lameck1) [inline src/modules/comments.js]: - [ ] We setup linters to help us write standardized code. Please enable linters and try fixing the issue involved. If it is a rule that proved hard to solve, you are only allowed to disable that particular rule and not disable linters for the whole file. Even so, code review will still have to figure out whether that rule you are disabling is something that can be easily solved or not.

File: src/modules/displayfood.js [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+import Comment from './comments.js';
+// import { modalDisplay } from './modules/commentpop.js';
+const main = document.querySelector('.main');
+
+const popup = document.querySelector('#modalDisplay');
+
+/* eslint-disable */
+export const display = async () => {
+/* eslint-enable */
```
Comment (Lameck1) [inline src/modules/displayfood.js]: - [ ] Same issue as discussed above, please enable linters and try solving the issues involved.

File: src/modules/food.js [hunk 3]
```diff
@@ -0,0 +1,2 @@
+/* eslint-disable */
```
Comment (Lameck1) [inline src/modules/food.js]: - [ ] Same issue as discussed above, please enable linters and try solving the issues involved.

File: src/modules/footer.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+const font = () => {
+  const footer = document.getElementById('footer');
+
+  footer.innerHTML += `<hr class="foot-line">
+<p class="foot-item">Created by microverse under CC license</p>
+`;
+};
+
+/* eslint-disable */
```
Comment (Lameck1) [inline src/modules/footer.js]: - [ ] Same issue as discussed above, please enable linters and try solving the issues involved.

File: src/modules/likes.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+/* eslint-disable */
```
Comment (Lameck1) [inline src/modules/likes.js]: - [ ] Same issue as discussed above, please enable linters and try solving the issues involved.

File: src/modules/displayfood.js [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+          const idss = data.filter((ids) => ids.item_id === id);
+          if (response.ok) {
+            kisplay(idss);
+          }
+        };
+        get(item.idMeal);
+        threeDiv.addEventListener('click', () => {
+          lik();
+          get(item.idMeal);
+        });
```
Comment (Lameck1) [inline src/modules/displayfood.js:84]: - [ ] We have one small bug that I believe you can solve. Let's take a look at the project requirement below:

  > When the user clicks on the Like button of an item, the interaction is recorded in the Involvement API and the screen is updated.

Right now when one clicks the like button, the screen is not updated. Instead, one has to refresh the page to notice that the like was added. Please let's refactor the code and see that this behavior is satisfied.

File: README.md [hunk 7]
```diff
@@ -1 +1,88 @@
...
+ Vscode. 
+ must have git installed.
+
+### Setup
+If you installed git you can clone the code to your machine, or download a ZIP of all the files directly.
+
+[Download the ZIP from this location], or run the following [git](https://git-scm.com/downloads) command to clone the files to your machine:
+
+```bash
+git clone  https://github.com/jaamanu/Javascript-Capstone-Project.git
```
Comment (Lameck1) [inline README.md:36]: - [ ] Please close your code blocks. The readme is not very readable without closing these code blocks. What this means is that a series of backticks`(```)` MUST be closed with a corresponding number of backticks `(```)` for it not to affect the next line text. Example

  image

File: README.md [hunk 8]
```diff
@@ -1 +1,88 @@
...
+- Once the files are on your machine, open the _todo-list_ folder in [Visual Studio Code](https://code.visualstudio.com/), and follow the steps in Installation & Set Up
+
+## 🛠 Installation & Set Up
+
+1. Install Node Modules
+
+   ```sh
+   npm install
```
Comment (Lameck1) [inline README.md:44]: - [ ] Same issue as above, please use the example below.

  image

File: README.md [hunk 9]
```diff
@@ -1 +1,88 @@
...
+   
+   2. Start the development server
+
+   ```sh
+   npm run start
```
Comment (Lameck1) [inline README.md:49]: - [ ] Same issue as above, please follow previous examples.

File: README.md [hunk 10]
```diff
@@ -1 +1,88 @@
...
+
+4. Run automated functionality test
+To check for functionality with automated test run npm test
+
+## 🚀 Building and Running for Production
+
+1. Generate a full static production build
+
+   ```sh
+   npm run build
```
Comment (Lameck1) [inline README.md:60]: - [ ] Same issue as above, please follow previous examples.

File: src/modules/APIs.js [hunk 11]
```diff
@@ -0,0 +1,17 @@
+import { foods } from './food.js';
+
+const commentkey = 'HJ3nJrQxKFWMKnh6RtwV';
+const commentUrl = `https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/${commentkey}/comments`;
+// const  = `${commentApi}/apps/${commentkey}/comments/`;
```
Comment (Lameck1) [inline src/modules/APIs.js]: - [ ] Please remove any line of code that is no longer useful.

File: src/modules/displayfood.js [hunk 12]
```diff
@@ -0,0 +1,192 @@
+import { foodApiObj, commentUrl, likeUrl } from './APIs.js';
+import Likes from './likes.js';
+
+import Comment from './comments.js';
+// import { modalDisplay } from './modules/commentpop.js';
```
Comment (Lameck1) [inline src/modules/displayfood.js]: - [ ] Please remove any line of code that is no longer useful.

**Review by Lameck1 — CHANGES_REQUESTED** [review]
Hi Team,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Clean code ✔️ 
- Good commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**Please, remember to tag me in your question so I can receive the notification.**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/displayfood.js (hunk 6 above)
Comment (jaamanu) [reply src/modules/displayfood.js:84]: https://www.loom.com/share/7d1f939852a2450c990e51a8490da1ab

**Comment by jaamanu** [comment]
Thank you for the review, i have made the necessary changes as requested, my apologies of deleting the README.md changes mistakenly in my VSCODE, i didnt think deleting in the vscode will also affect the PR comments you have made. 
image

File: src/modules/displayfood.js (hunk 6 above)
Comment (Lameck1) [reply src/modules/displayfood.js:84]: @jaamanu, sorry about this. I have realized that is some wait time/lag in the API post and get requests. This is all good now.

**Review by Lameck1 — APPROVED** [review]
Hi, Team,

Tasks for this project have been successfully implemented 💯.

## Highlights
- Professional README ✔️
- Clean code ✔️ 
- Good commit messages ✔️ 
- The MVP functions as expected.

All the best with the next project.

You are free to merge this branch :shipit:
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**Please, remember to tag me in your question so I can receive the notification.**

**Pull request merged**